Foundation/FileManager.swift

```diff
@@ -892,32 +884,43 @@ open class FileManager : NSObject {
         let ti = (TimeInterval(s.st_mtimespec.tv_sec) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_mtimespec.tv_nsec))
 #elseif os(Android)
         let ti = (TimeInterval(s.st_mtime) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_mtime_nsec))
+#elseif os(Windows)
+        let ti = (TimeInterval(s.st_mtime) - kCFAbsoluteTimeIntervalSince1970)
 #else
         let ti = (TimeInterval(s.st_mtim.tv_sec) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_mtim.tv_nsec))
 #endif
         result[.modificationDate] = Date(timeIntervalSinceReferenceDate: ti)
-        
+
+#if os(Windows)
+        result[.posixPermissions] = NSNumber(value: UInt64(s.st_mode & ~UInt16(ucrt.S_IFMT)))
```

Comment:
It might be worth adding an internal helper to mask `s.st_mode` to file permissions as it seems to be used in a few places.